libethcore/Precompiled.cpp

```diff
@@ -99,48 +99,77 @@ ETH_REGISTER_PRECOMPILED(identity)(bytesConstRef _in)
 
 // Parse _count bytes of _in starting with _begin offset as big endian int.
 // If there's not enough bytes in _in, consider it infinitely right-padded with zeroes.
-bigint parseBigEndianRightPadded(bytesConstRef _in, size_t _begin, size_t _count)
+bigint parseBigEndianRightPadded(bytesConstRef _in, bigint _begin, bigint _count)
 {
-	if (_begin > _in.count())
+	if (_begin > numeric_limits<size_t>::max())
```

Comment:
Hmm, seems like these two if's don't change much. Maybe having parameters as `bigint` would be enough?

if `_begin > numeric_limits<size_t>::max()` it won't pass the check `if (begin > _in.count())` just below.

Price calculation uses at most `_count=32` and if any length is greater than `numeric_limits<size_t>::max()` the price would be too big and we shouldn't get to implementation.